server/api/middleware.go

```diff
@@ -18,12 +18,59 @@ import (
 	"context"
 	"net/http"
 
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/log"
 	"github.com/tikv/pd/pkg/errs"
+	"github.com/tikv/pd/pkg/requestutil"
 	"github.com/tikv/pd/server"
 	"github.com/tikv/pd/server/cluster"
 	"github.com/unrolled/render"
+	"github.com/urfave/negroni"
+	"go.uber.org/zap"
 )
 
+// serviceInfoMiddleware is used to gather info from requsetInfo
+type serviceInfoMiddleware struct {
+	s                     *server.Server
+	registeredSericeLabel *requestutil.RequestSchemaList
+}
+
+func newServiceInfoMiddleware(s *server.Server) negroni.Handler {
+	registeredSericeLabel := requestutil.NewRequestSchemaList(5)
+
+	registeredSericeLabel.AddServiceLabel([]string{"pd", "api", "v1", "version"}, "GET", "GetPDVersion")
+	registeredSericeLabel.AddServiceLabel([]string{"pd", "api", "v1", "store", ""}, "GET", "ShowStore")
+	registeredSericeLabel.AddServiceLabel([]string{"pd", "api", "v1", "store", "", "state"}, "POST", "SetStoreState")
+	registeredSericeLabel.AddServiceLabel([]string{"pd", "api", "v1", "store", "", "label"}, "POST", "SetStoreLabel")
+	registeredSericeLabel.AddServiceLabel([]string{"pd", "api", "v1", "debug", "pprof", "profile"}, "", "DebugPprofProfile")
```

Comment:
Maybe better to put these configurations as a global variable.

Reply:
add magic string into const